Project/Assets/ML-Agents/Examples/Basic/Scripts/BasicAgent.cs

```diff
@@ -41,48 +49,33 @@ public override void AgentAction(float[] vectorAction)
         }
 
         m_Position += direction;
-        if (m_Position < m_MinPosition) { m_Position = m_MinPosition; }
-        if (m_Position > m_MaxPosition) { m_Position = m_MaxPosition; }
+        if (m_Position < k_MinPosition) { m_Position = k_MinPosition; }
+        if (m_Position > k_MaxPosition) { m_Position = k_MaxPosition; }
 
         gameObject.transform.position = new Vector3(m_Position - 10f, 0f, 0f);
 
-        AddReward(-0.01f);
+        m_Agent.AddReward(-0.01f);
 
-        if (m_Position == m_SmallGoalPosition)
+        if (m_Position == k_SmallGoalPosition)
         {
-            AddReward(0.1f);
-            Done();
+            m_Agent.AddReward(0.1f);
+            m_Agent.Done();
+            ResetAgent();
         }
 
-        if (m_Position == m_LargeGoalPosition)
+        if (m_Position == k_LargeGoalPosition)
         {
-            AddReward(1f);
-            Done();
+            m_Agent.AddReward(1f);
+            m_Agent.Done();
+            ResetAgent();
         }
     }
 
-    public override void AgentReset()
+    public void ResetAgent()
     {
         m_Position = 10;
-        m_MinPosition = 0;
-        m_MaxPosition = 20;
-        m_SmallGoalPosition = 7;
-        m_LargeGoalPosition = 17;
-        smallGoal.transform.position = new Vector3(m_SmallGoalPosition - 10f, 0f, 0f);
-        largeGoal.transform.position = new Vector3(m_LargeGoalPosition - 10f, 0f, 0f);
-    }
-
-    public override float[] Heuristic()
```

Comment:
Removed this since I don't want to inherit from Agent. I don't think it was getting us much.